### PR TITLE
Demet

### PR DESCRIPTION
**Pull request by recohen**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes
Added internal energy and smearing energy to analysis for pwscf in nexus
Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
tomcat3
## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- No. This PR is up to date with current the current state of 'develop'
  No. Code added or changed in the PR has been clang-formatted
No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by jtkrogel** [comment]
NB: it will take me a bit to get to this as I am (and others are) currently on a steep ramp for both proposal writing and March meeting.

**Comment by recohen** [comment]
No problem. Thanks! Ron

File: nexus/lib/pwscf_analyzer.py [hunk 1]
```diff
@@ -188,30 +214,11 @@ def analyze(self):
...
-            if self.info.warn:
-                self.warn('fermi energy read failed')
-            #end if
-        #end try
         try:
             energies = []
             for l in lines:
                 if l.find('!  ')!=-1:
-                    energies.append( float( l.split('=')[1].split()[0] ) )
+                    energies.append( eval( l.split('=')[1].split()[0] ) )
```
Comment (jtkrogel) [inline nexus/lib/pwscf_analyzer.py:221]: Avoid `eval` if at all possible (code injection vulnerability).  Does `float()` not work here?

File: nexus/lib/pwscf_analyzer.py [hunk 2]
```diff
@@ -188,30 +214,11 @@ def analyze(self):
             #end if
         #end try
```
Comment (jtkrogel) [inline nexus/lib/pwscf_analyzer.py:216]: Restore fermi energy processing.

File: nexus/lib/pwscf_analyzer.py [hunk 3]
```diff
@@ -242,17 +249,6 @@ def analyze(self):
             read_rel      = False
             for i in xrange(len(lines)):
                 l = lines[i]
```
Comment (jtkrogel) [inline nexus/lib/pwscf_analyzer.py:251]: Restore bands processing.

File: nexus/lib/pwscf_analyzer.py [hunk 4]
```diff
@@ -642,12 +638,7 @@ def analyze(self):
             try:
                 cont = self.input.control
                 datadir = os.path.join(self.path,cont.outdir,cont.prefix+'.save')
-                data_file = os.path.join(datadir,'data-file.xml')
-                if not os.path.exists(data_file):
-                    datadir = os.path.join(self.path,cont.outdir)
-                    data_file = os.path.join(datadir,cont.prefix+'.xml')
-                #end if
-                data = read_qexml(data_file)
+                data = read_qexml(os.path.join(datadir,'data-file.xml'))
```
Comment (jtkrogel) [inline nexus/lib/pwscf_analyzer.py:641]: Restore alternate xml data path.

File: nexus/lib/pwscf_analyzer.py [hunk 5]
```diff
@@ -930,6 +868,85 @@ def plot_bandstructure(self, filename=None, filepath=None, max_min_e = None, sho
             show()
         #end if
     #end def plot_bandstructure
+    def md_statistics(self,equil=None,autocorr=None):
```
Comment (jtkrogel) [inline nexus/lib/pwscf_analyzer.py:871]: Add `#end` closures.

Comment (jtkrogel) [inline nexus/lib/pwscf_analyzer.py:871]: `md_statistics` and `md_plots` appear to be repeated with some variations in a few files.  Is there a good way to consolidate and avoid duplication?

File: nexus/lib/pwscf_input.py [hunk 6]
```diff
@@ -906,7 +906,6 @@ class ee(Section):
 #end for
 
 
-exit_ = exit
```
Comment (jtkrogel) [inline nexus/lib/pwscf_input.py:909]: Restore alias for `exit` function.

**Comment by jtkrogel** [comment]
Note: the updates here should be checked against the test added in #2362.

**Comment by prckent** [comment]
Closing this as stale. Would be happy to review a freshened and up to date version. I would add any utilities separately from new machine support. An example of adding a new machine that was merged is at https://github.com/QMCPACK/qmcpack/pull/4598 .

**Pull request closed without merging**